caron.js

```diff
@@ -208,6 +209,9 @@ class Caron {
         'local jopt0 = cmsg["$removeOnComplete"]',
         'if (type(jopt0) ~= "boolean") then jopt0 = true end',
         'local jopts = { removeOnComplete = jopt0, attempts = jattempts, delay = jdelay }',
+        'if (type(jbackoff) == "string") then', 
+        '  jopts["backoff"] = { type = jbackoff, delay = jdelay }',
+        'end',
```

Comment:
```suggestion
        'end',
        'cmsg["$backoff"] = nil',
```
we need to remove the special key from the event data